src/main/java/teammates/ui/controller/InstructorCommentsPageData.java

```diff
@@ -146,6 +120,31 @@ public boolean isViewingDraft() {
         return isViewingDraft;
     }        
 
+    private String getRecipientNames(Set<String> recipients) {
+        StringBuilder namesStringBuilder = new StringBuilder();
+        int i = 0;
+        for (String recipient : recipients) {
+            if (i == recipients.size() - 1 && recipients.size() > 1) {
+                namesStringBuilder.append("and ");
+            }
+            StudentAttributes student = roster.getStudentForEmail(recipient);
+            if (courseId.equals(recipient)) { 
+                namesStringBuilder.append("<b>All students in this course</b>, ");
+            } else if (student != null) {
+                if (recipients.size() == 1) {
+                    namesStringBuilder.append("<b>" + student.name + " (" + student.team + ", " + student.email + ")</b>, ");
+                } else {
+                    namesStringBuilder.append("<b>" + student.name + "</b>" + ", ");
+                }
+            } else {
+                namesStringBuilder.append("<b>" + recipient + "</b>" + ", ");
+            }
+            i++;
+        }
+        String namesString = namesStringBuilder.toString();
+        return removeEndComma(namesString);
+    }
```

Comment:
It is better if this method takes the roster as a parameter. When possible, feed everything the method needs as parameters. That way they are easier to test.